src/util/triplea/MapXMLCreator/TerritoryConnectionsPanel.java

```diff
@@ -0,0 +1,256 @@
+package util.triplea.MapXMLCreator;
+
+import games.strategy.util.AlphanumComparator;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Polygon;
+import java.awt.Shape;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Area;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import util.image.ConnectionFinder;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class TerritoryConnectionsPanel extends ImageScrollPanePanel
+{
+	
+	String m_selectedTerritory = null;
+
+	private TerritoryConnectionsPanel()
+	{
+		s_instance = this;
+	}
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel)
+	{
+		s_mapXMLCreator = mapXMLCreator;
+		final TerritoryConnectionsPanel panel = new TerritoryConnectionsPanel();
+		panel.layout(stepActionPanel);
+		mapXMLCreator.setAutoFillAction(new AbstractAction()

```

Comment:
This way it is more flexible to for the calling method to access any variable it is currently handling. It would be much more complex to try to move the logic of each panel to MapXMLCreator and it would not belong there.